base/regex.jl

```diff
@@ -174,6 +174,62 @@ function occursin(r::Regex, s::SubString; offset::Integer=0)
                      r.match_data)
 end
 
+"""
+    startswith(s::AbstractString, prefix::Regex)
+
+Return `true` if `s` starts with the Regex pattern, `prefix`.
+
+See also [`endswith`](@ref).
+
+# Examples
+```jldoctest
+julia> startswith("JuliaLang", r"Julia|Romeo")
+true
+```
+"""
+function startswith(s::AbstractString, r::Regex)
+    compile(r)
+    return PCRE.exec(r.regex, String(s), 0, r.match_options | PCRE.ANCHORED,
+                     r.match_data)
+end
+
+function startswith(s::SubString, r::Regex)
+    compile(r)
+    return PCRE.exec(r.regex, s, 0, r.match_options | PCRE.ANCHORED,
+                     r.match_data)
+end
+    rr = Regex(r.pattern, r.compile_options | PCRE.ANCHORED, r.match_options)
+    return occursin(rr, s)
+end
+
+"""
+    endswith(s::AbstractString, suffix::Regex)
+
+Return `true` if `s` ends with the Regex pattern, `suffix`.
+
+See also [`startswith`](@ref).
+
+# Examples
+```jldoctest
+julia> endswith("JuliaLang", r"Lang|Roberts")
+true
+```
+"""
+function endswith(s::AbstractString, r::Regex)
+    compile(r)
+    return PCRE.exec(r.regex, String(s), 0, r.match_options | PCRE.ENDANCHORED,
+                     r.match_data)
+end
+
+function endswith(s::SubString, r::Regex)
+    compile(r)
+    return PCRE.exec(r.regex, s, 0, r.match_options | PCRE.ENDANCHORED,
+                     r.match_data)
+end
+    rr = Regex(r.pattern, r.compile_options | PCRE.ENDANCHORED, r.match_options)
```

Comment:
The lines 201-203 and 229-231 need to be deleted (trailing garbage in the function definitions; currently, this shouldn't even parse). 

The mistake crept in because I failed at using the github "suggest [beta]" feature correctly, which produced a bad diff. Sorry.

Reply:
I  will fix it today.  The feature did mention it was in "Beta", though. :smile: